### ISSUE TITLE
[FE] 10-10CG - File Upload: [Assistive tech and device support] Focus state isn't visible. (11.09.2)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#27411]
#### VFS team name
Caregivers

#### VFS product name
Improve File Upload Error Messaging

#### Point of Contact/Reviewers
Angela Fowler

---

### Platform Issue
Focus state isn't visible.

### Issue Details
When an error is encountered when attempting to upload a file, focus is not placed at the point of the error. 

**NOTE: This issue may already be resolved.  Please check the steps and confirm**

### Supporting Guidance
[11.09](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 2.4.7_AA

---

### Platform Recommendation
When the user encounters an error, focus should be placed on the try again button or next relevant field. The error should be linked to said button or field using `aria-describedby` as documented in the VA Design System. https://design.va.gov/patterns/form-feedback

**NOTE: This issue may already be resolved.  Please check the steps and confirm**

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] VFS team closes the ticket when the issue has been resolved

**Comment by hdjustice** [comment]
In staging environment, I ran a test to upload a .txt file.  The error message appeared and the focus is on the "Upload a new file" button.  I will have the team check this out, it may already be resolved.
cc: @markfallows

**Comment by longmd** [comment]
@hdjustice , this one goes back to the forms system team as well. This form is utilizing their FileUploadUI component.

As far as the behavior I'm seeing, focus is given to the file list on error, which contains a list item that has the file name, error message and "delete file" button. The first tab instance then gives focus to the delete button. However, the `aria-describedby` on that button is linked to the container with only the file name and not tied to the error message container. The error message container does have `role="alert"` but VoiceOver on Mac does not read the message.

**Comment by hdjustice** [comment]
@shiragoodman - This is an older ticket (last July!) but it seems that the File Upload component we are using isn't allowing us to make any changes that would improve the accessibility here.  I am not sure who should take a look at this.  Can you help me?

Here is the initial feedback from Angela:
> When the user encounters an error, focus should be placed on the try again button or next relevant field. The error should be linked to said button or field using aria-describedby as documented in the VA Design System.

Here is the feedback from Matt (Frontend) on this issue
> As far as the behavior I'm seeing, focus is given to the file list on error, which contains a list item that has the file name, error message and "delete file" button. The first tab instance then gives focus to the delete button. However, the aria-describedby on that button is linked to the container with only the file name and not tied to the error message container. The error message container does have role="alert" but VoiceOver on Mac does not read the message.

**Comment by briandeconinck** [comment]
Yeah, this sounds like it's probably a component issue for the Design System team to look at. And I think @joshkimux might also be tracking another file upload issue that might be component related? Tagging him in case it's similar or related behavior to what was reported here.

**Comment by shiragoodman** [comment]
FYI @k80bowman @caw310 

Adding the Design System team label for you all to take a look. Thanks!

**Comment by k80bowman** [comment]
We're working on a File input web component right now, so I will make sure this gets addressed.
